planner/core/common_plans.go

```diff
@@ -761,18 +761,20 @@ type Delete struct {
 
 // AnalyzeTableID is hybrid table id used to analyze table.
 type AnalyzeTableID struct {
-	PersistID  int64
-	CollectIDs []int64
+	PersistID int64
+	// FatherID just used in the partition table.
+	// It represents the ID of the table to which the partition belongs.
+	FatherID int64
 }
```

Comment:
Give an example to make the meaning of these IDs more clear.
Version1: no global-level stats
Version2: only global-level stats
Version3(This PR): both have partition-level stats and global-level stats, and the global-level stats are merged from the partition-level stats

Now, we have table `t`(ID = 1). And the table `t` has three partitions `p1`(ID = 2), `p2`(ID = 3), `p3`(ID = 4).

In version1, the `analyze` request has three tasks: `AnalyzeTableID{PersistID = 2, CollectIDs = {2}}`, `AnalyzeTableID{PersistID = 3, CollectIDs = {3}}` , `AnalyzeTableID{PersistID = 4, CollectIDs = {4}}`.

In version2, the `analyze` request has only one task: `AnalyzeTableID{PersistID = 1, CollectIDs = {2, 3, 4}}`.

In version3, the `analyze` request has three tasks: `AnalyzeTableID{PersistID = 2, FatherID= 1}`, `AnalyzeTableID{PersistID = 3, FatherID= 1}` , `AnalyzeTableID{PersistID = 4, FatherID= 1}`.